warcprox/crawl_log.py

```diff
@@ -64,13 +64,38 @@ def notify(self, recorded_url, records):
         else:
             content_length = 0
             payload_digest = '-'
+        logging.info('warcprox_meta %s' , recorded_url.warcprox_meta)
+
+        hop_path = recorded_url.warcprox_meta.get('metadata', {}).get('hop_path')
+        brozzled_url = recorded_url.warcprox_meta.get('metadata', {}).get('brozzled_url')
+        hop_via_url = recorded_url.warcprox_meta.get('metadata', {}).get('hop_via_url')
+
+        if hop_path is None and brozzled_url is None and hop_via_url is None:
+            #No hop info headers provided
+            hop_path = "-"
+            via_url = recorded_url.referer or '-'
+        else:
+            if hop_path is None:
+                hop_path = "-"
+            if hop_via_url is None:
+                hop_via_url = "-"
+            #Prefer referer header. Otherwise use provided via_url
+            via_url = recorded_url.referer or hop_via_url if hop_path != "-" else "-"
+            if brozzled_url != recorded_url.url.decode('ascii') and "brozzled_url" in recorded_url.warcprox_meta.get('metadata', {}).keys():
```

Comment:
could `recorded_url.url` contain non-ascii chars?

Reply:
This did uncover a bug. recorded_url.url is guaranteed to be ascii since http enforces ascii, but the two other urls we are sending in the json encoded header are not similarly encoded, and need to be for proper comparison.